{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# make revision base file\n",
    "import sys, os\n",
    "\n",
    "input_file = file(\"../CpG/v10_CpG_CtoT.txt\")\n",
    "temp_file = file(\"../CpG/temp.txt\")\n",
    "output_file = file(\"revision_CpG_CtoT.txt\",\"w\")\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "temp_line = temp_file.readline().strip()\n",
    "input_split = temp_line.split('\\t')\n",
    "output_file.write(input_split[0] + '\\t' + input_split[1] + '\\t' +input_split[2] + '\\t' +input_split[3] + '\\t' + input_split[4] + '\\t' + input_split[9] + '\\t' + input_split[10] + '\\t' + input_split[11] + '\\t' + input_split[12] + '\\t' + input_split[13] + '\\t' + input_split[14] + '\\t' + input_split[16] + '\\t' + input_split[17] + '\\t' + input_split[18] + '\\t' + input_split[19] + '\\t' + input_split[20] + '\\t' + input_split[26].replace(\",\",\"\") + '\\t' + input_split[27].replace(\",\",\"\") + '\\t' +input_split[28] + '\\t' +input_split[33] + ';'+input_split[34] + ';'+input_split[35] + '\\n')\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    output_file.write(input_split[0] + '\\t' + input_split[1] + '\\t' +input_split[2] + '\\t' +input_split[3] + '\\t' + input_split[4] + '\\t' + input_split[9] + '\\t' + input_split[10] + '\\t' + input_split[11] + '\\t' + input_split[12] + '\\t' + input_split[13] + '\\t' + input_split[14] + '\\t' + input_split[16] + '\\t' + input_split[17] + '\\t' + input_split[18] + '\\t' + input_split[19] + '\\t' + input_split[20] + '\\t' + input_split[26].replace(\",\",\"\") + '\\t' + input_split[27].replace(\",\",\"\") + '\\t' +input_split[28] + '\\t' +input_split[33] + ';'+input_split[34] + ';'+input_split[35] + '\\n')\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr2\n",
      "chr2\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "invalid literal for int() with base 10: '##fileformat=VCFv4.1'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-48-7985a1d4b69b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m    137\u001b[0m     \u001b[0;31m#thousand genome annotation\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    138\u001b[0m     \u001b[0minfo_split\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0minfo_line\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 139\u001b[0;31m     \u001b[0minfo_chr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minfo_split\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"chr\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"X\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"23\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreplace\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Y\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"24\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    140\u001b[0m     \u001b[0minfo_pos\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minfo_split\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: invalid literal for int() with base 10: '##fileformat=VCFv4.1'"
     ]
    }
   ],
   "source": [
    "#add thousand genome and PCAWG data including individual genotypes\n",
    "\n",
    "import sys, os, gzip\n",
    "input_file = file(\"revision_CpG_CtoT.txt\")\n",
    "output_file = file(\"revision_CpG_CtoT_PCAWG_individual.txt\",\"w\")\n",
    "output_file1 = file(\"revision_CpG_CtoT_PCAWG.txt\",\"w\")\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "info_file = gzip.open(\"/home/users/jhyouk/89_backup_Workstation/jhyouk/CpG_revision/1000Genomes/ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\")\n",
    "pcawg_file = gzip.open(\"/home/users/jhyouk/89_backup_Workstation/jhyouk/CpG_revision/PCAWG-8/pcawg8.snps.indels.svs.phased.icgc.v2.vcf.gz\")\n",
    "info_line = info_file.readline().strip()\n",
    "pcawg_line = pcawg_file.readline().strip()\n",
    "while pcawg_line[0:2] == '##':\n",
    "    pcawg_line = pcawg_file.readline().strip()\n",
    "while info_line[0:2] == '##':\n",
    "    info_line = info_file.readline().strip()\n",
    "    \n",
    "info_chr = 1\n",
    "\n",
    "pcawg_split = pcawg_line.split('\\t')\n",
    "info_split = info_line.split('\\t')\n",
    "\n",
    "output_file.write(input_line + '\\tPCAWG\\tThousand\\tTotal_Af;AC;AN\\tRace_thousand_EAS;AMR;AFR;EUR;SAS\\tGT_PCAWG;Thousand\\t' + '\\t'.join(pcawg_split[9:]) + '\\t'+  '\\t'.join(info_split[9:]) +'\\n')\n",
    "output_file1.write(input_line + '\\tPCAWG\\tThousand\\tTotal_Af;AC;AN\\tRace_thousand_EAS;AMR;AFR;EUR;SAS\\tGT_PCAWG;Thousand' +'\\n')\n",
    "\n",
    "input_line = input_file.readline().strip()\n",
    "info_line = info_file.readline().strip()\n",
    "pcawg_line = pcawg_file.readline().strip()\n",
    "p_done = 'no'\n",
    "nn=0\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_chr = int(input_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "    input_pos = int(input_split[1])\n",
    "    input_ref = input_split[3]\n",
    "    input_alt = input_split[4]\n",
    "    \n",
    "    p_AC = 0\n",
    "    p_AN = 0\n",
    "    \n",
    "    t_info = '.'\n",
    "    race_info = '.'\n",
    "    \n",
    "    gt_pcawg = 0\n",
    "    gt_thousand = 0   \n",
    "    \n",
    "    \n",
    "    p_indi = []\n",
    "    for i in range(0,1823):\n",
    "        p_indi.append('.')\n",
    "    t_indi = []\n",
    "    for i in range(0,2504):\n",
    "        t_indi.append('.')    \n",
    "    \n",
    "    \n",
    "    if input_chr == info_chr:\n",
    "        'blank'\n",
    "    else:\n",
    "        #nn+=1\n",
    "        print input_split[0]\n",
    "        info_file.close()\n",
    "        if input_chr == 'chrX':\n",
    "            info_file = gzip.open(\"/home/users/jhyouk/89_backup_Workstation/jhyouk/CpG_revision/1000Genomes/ALL.chrX.phase3_shapeit2_mvncall_integrated_v1b.20130502.genotypes.vcf.gz\")\n",
    "        elif input_chr == 'chrY':\n",
    "            info_file = gzip.open(\"/home/users/jhyouk/89_backup_Workstation/jhyouk/CpG_revision/1000Genomes/ALL.chrY.phase3_integrated_v2a.20130502.genotypes.vcf.gz\")     \n",
    "        else:\n",
    "            info_file = gzip.open(\"/home/users/jhyouk/89_backup_Workstation/jhyouk/CpG_revision/1000Genomes/ALL.%s.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\" % input_split[0])   \n",
    "        info_chr = input_chr\n",
    "        \n",
    "        info_line = info_file.readline().strip()\n",
    "        while info_line[0:2] == '##':\n",
    "            info_line = info_file.readline().strip()\n",
    "        info_line = info_file.readline().strip()\n",
    "        \n",
    "    \n",
    "    if p_done == 'no':\n",
    "        p_info = '.'\n",
    "        #PCAWG annotation\n",
    "        p_split = pcawg_line.split('\\t')\n",
    "        p_chr = int(p_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "        p_pos = int(p_split[1])\n",
    "        #dec_target = 'none'\n",
    "        if p_chr < input_chr:\n",
    "            pcawg_line = pcawg_file.readline().strip()\n",
    "            continue\n",
    "        elif p_chr == input_chr:\n",
    "            if p_pos < input_pos:\n",
    "                pcawg_line = pcawg_file.readline().strip()\n",
    "                continue\n",
    "            elif p_pos > input_pos:\n",
    "                p_done = 'yes'\n",
    "                p_info = '.;.;.'\n",
    "                #gt_pcawg = 0\n",
    "            else:\n",
    "                p_ref = p_split[3]\n",
    "                p_alt = p_split[4].split(',')\n",
    "\n",
    "                p_target = 0\n",
    "                dec_target = '.'\n",
    "                if p_ref == 'C':\n",
    "                    for p in p_alt:\n",
    "                        if p == 'T':\n",
    "                            dec_target = 'bingo'\n",
    "                            break\n",
    "                        else:\n",
    "                            p_target+=1\n",
    "                elif p_ref == 'G':\n",
    "                    for p in p_alt:\n",
    "                        if p == 'A':\n",
    "                            dec_target = 'bingo'\n",
    "                            break\n",
    "                        else:\n",
    "                            p_target+=1\n",
    "                else:\n",
    "                    'blank'\n",
    "\n",
    "                if dec_target == 'bingo':\n",
    "                    gt_pcawg = p_target + 1\n",
    "                    p_num = p_split[7].split(';')\n",
    "                    for i in p_num:\n",
    "                        if i[0:3] == 'AN=':\n",
    "                            p_AN = float(i.split('=')[1])\n",
    "                        elif i[0:3] == 'AC=':\n",
    "                            temp_AC = i.split('=')[1].split(',')\n",
    "                            p_AC = float(temp_AC[p_target])\n",
    "                    p_info = '%s;%s;%s' %(round(p_AC/p_AN,5),p_AC,p_AN)\n",
    "                    p_indi = p_split[9:]\n",
    "                    #nn+=1\n",
    "                else:\n",
    "                    p_info = '.;.;.'\n",
    "                    \n",
    "                    \n",
    "                p_done = 'yes'\n",
    "                pcawg_line = pcawg_file.readline().strip()\n",
    "\n",
    "        else:\n",
    "            p_done = 'yes'\n",
    "            p_info = '.;.;.'\n",
    "            \n",
    "    else: #p_done == 'yes\n",
    "        'blank'\n",
    "    \n",
    "    #thousand genome annotation\n",
    "    info_split = info_line.split('\\t')\n",
    "    info_chr = int(info_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "    info_pos = int(info_split[1])\n",
    "    \n",
    "    info_AC = 0\n",
    "    info_AN = 0\n",
    "    info_race = '.'\n",
    "    af_race = ['0','0','0','0','0']\n",
    "\n",
    "    info_target = 'none'\n",
    "    if info_chr < input_chr:\n",
    "        info_line = info_file.readline().strip()\n",
    "        continue\n",
    "    elif info_chr == input_chr:\n",
    "        if info_pos < input_pos:\n",
    "            info_line = info_file.readline().strip()\n",
    "            continue            \n",
    "        elif info_pos > input_pos:\n",
    "            t_info = '.;.;.'\n",
    "            info_race = '.;.;.;.;.'\n",
    "        else:\n",
    "            info_ref = info_split[3]\n",
    "            info_alt = info_split[4].split(',')\n",
    "            \n",
    "            t_target =0\n",
    "            t_decision = '.'\n",
    "            if info_ref == 'C':\n",
    "                for t in info_alt:\n",
    "                    if t == 'T':\n",
    "                        t_decision = 'bingo'\n",
    "                        break\n",
    "                    else:\n",
    "                        t_target+=1\n",
    "            elif info_ref == 'G':\n",
    "                for t in info_alt:\n",
    "                    if t == 'A':\n",
    "                        t_decision = 'bingo'\n",
    "                        break\n",
    "                    else:\n",
    "                        t_target +=1\n",
    "            else:\n",
    "                'blank'\n",
    "            \n",
    "            if t_decision == 'bingo':\n",
    "                gt_thousand = t_target + 1\n",
    "                t_num = info_split[7].split(';')\n",
    "                for i in t_num:\n",
    "                    #print input_line\n",
    "                    #print i\n",
    "                    if i[0:3] == 'AN=':\n",
    "                        info_AN = float(i.split('=')[1])\n",
    "                    elif i[0:3] == 'AC=':\n",
    "                        temp_AC = i.split('=')[1].split(',')\n",
    "                        info_AC = float(temp_AC[t_target])\n",
    "                    elif i[0:3] == 'EAS':\n",
    "                        temp= i.split('=')[1].split(',')\n",
    "                        af_race[0] = str(temp[t_target])                        \n",
    "                    elif i[0:3] == 'AMR':\n",
    "                        temp= i.split('=')[1].split(',')\n",
    "                        af_race[1] = str(temp[t_target])                          \n",
    "                    elif i[0:3] == 'AFR':\n",
    "                        temp= i.split('=')[1].split(',')\n",
    "                        af_race[2] = str(temp[t_target])                          \n",
    "                    elif i[0:3] == 'EUR':\n",
    "                        temp= i.split('=')[1].split(',')\n",
    "                        af_race[3] = str(temp[t_target])  \n",
    "                    elif i[0:3] == 'SAS':\n",
    "                        temp= i.split('=')[1].split(',')\n",
    "                        af_race[4] = str(temp[t_target])  \n",
    "                        \n",
    "                t_info = '%s;%s;%s' %(round(info_AC/info_AN,5),info_AC,info_AN)   \n",
    "                info_race = ';'.join(af_race)\n",
    "                #print '\\t'.join(info_split[9:15])\n",
    "                t_indi = info_split[9:]\n",
    "                \n",
    "            else:\n",
    "                t_info = '.;.;.'\n",
    "                info_race = '.;.;.;.;.'\n",
    "                \n",
    "    else:\n",
    "        t_info = '.;.;.'\n",
    "        info_race = '.;.;.;.;.'\n",
    "    \n",
    "    total_info = '.'\n",
    "    total_AC = float(p_AC + info_AC)\n",
    "    total_AN = float(p_AN + info_AN)\n",
    "    total_af = 0\n",
    "    if total_AN == 0:\n",
    "        total_info = '.;.;.'\n",
    "    else:\n",
    "        total_af = round(float(total_AC/total_AN),5)\n",
    "        total_info = str(total_af) + ';' + str(total_AC) + ';' + str(total_AN)\n",
    "    \n",
    "    \n",
    "    output_file.write(input_line + '\\t' + p_info +'\\t' + t_info + '\\t' + total_info + '\\t' + info_race + '\\t' + str(gt_pcawg) + ';' + str(gt_thousand) + '\\t' + '\\t'.join(p_indi) + '\\t' + '\\t'.join(t_indi) + '\\n')\n",
    "    output_file1.write(input_line + '\\t' + p_info +'\\t' + t_info + '\\t' + total_info + '\\t' + info_race + '\\t' + str(gt_pcawg) + ';' + str(gt_thousand) + '\\n')\n",
    "    p_done = 'no'\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    #nn+=1\n",
    "    #if nn>1:\n",
    "        #break\n",
    "\n",
    "output_file.close()\n",
    "output_file1.close()\n",
    "info_file.close()\n",
    "pcawg_file.close()\n",
    "input_file.close()\n",
    "\n",
    "print 'THE END'\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1823\n"
     ]
    }
   ],
   "source": [
    "import gzip\n",
    "pcawg_file = gzip.open(\"/home/users/jhyouk/89_backup_Workstation/jhyouk/CpG_revision/PCAWG-8/pcawg8.snps.indels.svs.phased.icgc.v2.vcf.gz\")\n",
    "pcawg_line = pcawg_file.readline().strip()\n",
    "while pcawg_line[0:2] == '##':\n",
    "    pcawg_line = pcawg_file.readline().strip()\n",
    "pcawg_split = pcawg_line.split('\\t')\n",
    "print len(pcawg_split[9:])\n",
    "#print '\\t'.join(pcawg_split[9:])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2504\n"
     ]
    }
   ],
   "source": [
    "import gzip\n",
    "info_file = gzip.open(\"/home/users/jhyouk/89_backup_Workstation/jhyouk/CpG_revision/1000Genomes/ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\")\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "while info_line[0:2] == '##':\n",
    "    info_line = info_file.readline().strip()\n",
    "info_split = info_line.split('\\t')\n",
    "print len(info_split[9:])\n",
    "#'\\t'.join(info_split[9:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1823\n",
      "['.', '.', '.', '.', '.', '.', '.', '.', '.', '.']\n"
     ]
    }
   ],
   "source": [
    "p_temp = []\n",
    "for i in range(0,1823):\n",
    "    p_temp.append('.')\n",
    "print len(p_temp)\n",
    "print p_temp[0:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4352\n",
      "4352\n"
     ]
    }
   ],
   "source": [
    "input_file = file(\"revision_CpG_CtoT_PCAWG.txt\")\n",
    "input_line = input_file.readline().strip()\n",
    "m=0\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    print len(input_split)\n",
    "    m+=1\n",
    "    if m>1:\n",
    "        break\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t1247977\t1247977\tC\tT\tchr1:1247918\t-229.0\to\t575\t575\torphan\texonic\tCPSF3L\tsynonymous SNV\tCPSF3L:NM_001256463:exon11:c.G1095A:p.A365A,CPSF3L:NM_001256456:exon15:c.G1416A:p.A472A,CPSF3L:NM_001256460:exon14:c.G1311A:p.A437A,CPSF3L:NM_001256462:exon11:c.G1104A:p.A368A,CPSF3L:NM_017871:exon13:c.G1398A:p.A466A,\t0.842105\tNR_106785\tMIR6727\tNR_TSS\t2.491e-05;3;120430\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr1\t1247978\t1247978\tG\tA\tchr1:1247918\t-228.0\to\t575\t575\torphan\texonic\tCPSF3L\tnonsynonymous SNV\tCPSF3L:NM_001256463:exon11:c.C1094T:p.A365V,CPSF3L:NM_001256456:exon15:c.C1415T:p.A472V,CPSF3L:NM_001256460:exon14:c.C1310T:p.A437V,CPSF3L:NM_001256462:exon11:c.C1103T:p.A368V,CPSF3L:NM_017871:exon13:c.C1397T:p.A466V,\t0.842105\tNR_106785\tMIR6727\tNR_TSS\t1.411e-04;17;120456\t.;.;.\t0.0006;3.0;5008.0\t0.0006;3.0;5008.0\t0;0;0.0023;0;0\t0;1\n",
      "chr1\t1247987\t1247987\tC\tT\tchr1:1247918\t-219.0\to\t575\t575\torphan\texonic\tCPSF3L\tnonsynonymous SNV\tCPSF3L:NM_001256463:exon11:c.G1085A:p.R362Q,CPSF3L:NM_001256456:exon15:c.G1406A:p.R469Q,CPSF3L:NM_001256460:exon14:c.G1301A:p.R434Q,CPSF3L:NM_001256462:exon11:c.G1094A:p.R365Q,CPSF3L:NM_017871:exon13:c.G1388A:p.R463Q,\t0.826087\tNR_106785\tMIR6727\tNR_TSS\t5.806e-05;7;120574\t.;.;.\t0.0002;1.0;5008.0\t0.0002;1.0;5008.0\t0;0;0.0008;0;0\t0;1\n",
      "chr1\t1247988\t1247988\tG\tA\tchr1:1247918\t-218.0\to\t575\t575\torphan\texonic\tCPSF3L\tnonsynonymous SNV\tCPSF3L:NM_001256463:exon11:c.C1084T:p.R362W,CPSF3L:NM_001256456:exon15:c.C1405T:p.R469W,CPSF3L:NM_001256460:exon14:c.C1300T:p.R434W,CPSF3L:NM_001256462:exon11:c.C1093T:p.R365W,CPSF3L:NM_017871:exon13:c.C1387T:p.R463W,\t0.826087\tNR_106785\tMIR6727\tNR_TSS\t1.659e-05;2;120578\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr1\t1248004\t1248004\tC\tT\tchr1:1247918\t-202.0\to\t575\t575\torphan\texonic\tCPSF3L\tsynonymous SNV\tCPSF3L:NM_001256463:exon11:c.G1068A:p.S356S,CPSF3L:NM_001256456:exon15:c.G1389A:p.S463S,CPSF3L:NM_001256460:exon14:c.G1284A:p.S428S,CPSF3L:NM_001256462:exon11:c.G1077A:p.S359S,CPSF3L:NM_017871:exon13:c.G1371A:p.S457S,\t0.9\tNR_106785\tMIR6727\tNR_TSS\t1.657e-05;2;120714\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr1\t1248005\t1248005\tG\tA\tchr1:1247918\t-201.0\to\t575\t575\torphan\texonic\tCPSF3L\tnonsynonymous SNV\tCPSF3L:NM_001256463:exon11:c.C1067T:p.S356L,CPSF3L:NM_001256456:exon15:c.C1388T:p.S463L,CPSF3L:NM_001256460:exon14:c.C1283T:p.S428L,CPSF3L:NM_001256462:exon11:c.C1076T:p.S359L,CPSF3L:NM_017871:exon13:c.C1370T:p.S457L,\t0.9\tNR_106785\tMIR6727\tNR_TSS\t3.313e-05;4;120728\t.;.;.\t0.0002;1.0;5008.0\t0.0002;1.0;5008.0\t0.001;0;0;0;0\t0;1\n",
      "chr1\t1248015\t1248015\tC\tT\tchr1:1247918\t-191.0\to\t575\t575\torphan\texonic\tCPSF3L\tnonsynonymous SNV\tCPSF3L:NM_001256463:exon11:c.G1057A:p.V353I,CPSF3L:NM_001256456:exon15:c.G1378A:p.V460I,CPSF3L:NM_001256460:exon14:c.G1273A:p.V425I,CPSF3L:NM_001256462:exon11:c.G1066A:p.V356I,CPSF3L:NM_017871:exon13:c.G1360A:p.V454I,\t0.964286\tNR_106785\tMIR6727\tNR_TSS\t8.279e-06;1;120792\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr1\t1248016\t1248016\tG\tA\tchr1:1247918\t-190.0\to\t575\t575\torphan\texonic\tCPSF3L\tsynonymous SNV\tCPSF3L:NM_001256463:exon11:c.C1056T:p.P352P,CPSF3L:NM_001256456:exon15:c.C1377T:p.P459P,CPSF3L:NM_001256460:exon14:c.C1272T:p.P424P,CPSF3L:NM_001256462:exon11:c.C1065T:p.P355P,CPSF3L:NM_017871:exon13:c.C1359T:p.P453P,\t0.964286\tNR_106785\tMIR6727\tNR_TSS\t8.278e-06;1;120804\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr1\t1248040\t1248040\tC\tT\tchr1:1247918\t-166.0\to\t575\t575\torphan\texonic\tCPSF3L\tsynonymous SNV\tCPSF3L:NM_001256463:exon11:c.G1032A:p.T344T,CPSF3L:NM_001256456:exon15:c.G1353A:p.T451T,CPSF3L:NM_001256460:exon14:c.G1248A:p.T416T,CPSF3L:NM_001256462:exon11:c.G1041A:p.T347T,CPSF3L:NM_017871:exon13:c.G1335A:p.T445T,\t0.95\tNR_106785\tMIR6727\tNR_TSS\t4.964e-05;6;120868\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr1\t1248041\t1248041\tG\tA\tchr1:1247918\t-165.0\to\t575\t575\torphan\texonic\tCPSF3L\tnonsynonymous SNV\tCPSF3L:NM_001256463:exon11:c.C1031T:p.T344M,CPSF3L:NM_001256456:exon15:c.C1352T:p.T451M,CPSF3L:NM_001256460:exon14:c.C1247T:p.T416M,CPSF3L:NM_001256462:exon11:c.C1040T:p.T347M,CPSF3L:NM_017871:exon13:c.C1334T:p.T445M,\t0.95\tNR_106785\tMIR6727\tNR_TSS\t2.895e-04;35;120882\t.;.;.\t0.001;5.0;5008.0\t0.001;5.0;5008.0\t0;0.0014;0.003;0;0\t0;1\n",
      "chr1\t1248046\t1248046\tC\tT\tchr1:1247918\t-160.0\to\t575\t575\torphan\texonic\tCPSF3L\tsynonymous SNV\tCPSF3L:NM_001256463:exon11:c.G1026A:p.T342T,CPSF3L:NM_001256456:exon15:c.G1347A:p.T449T,CPSF3L:NM_001256460:exon14:c.G1242A:p.T414T,CPSF3L:NM_001256462:exon11:c.G1035A:p.T345T,CPSF3L:NM_017871:exon13:c.G1329A:p.T443T,\t0.95\tNR_106785\tMIR6727\tNR_TSS\t9.926e-05;12;120898\t.;.;.\t0.0002;1.0;5008.0\t0.0002;1.0;5008.0\t0;0;0.0008;0;0\t0;1\n",
      "chr2\t264969\t264969\tC\tT\tchr2:263400\t-649.5\to\t1838\t1838\tTSS\texonic\tACP1\tnonsynonymous SNV\tACP1:NM_007099:exon1:c.C5T:p.A2V,ACP1:NM_001040649:exon1:c.C5T:p.A2V,ACP1:NM_004300:exon1:c.C5T:p.A2V,\t0\tNR_104225\tSH3YL1\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr3\t4345060\t4345060\tC\tT\tchr3:4345005\t-114.0\to\t337\t337\torphan\texonic\tSETMAR\tsynonymous SNV\tSETMAR:NM_001276325:exon1:c.C6T:p.F2F,SETMAR:NM_001243723:exon1:c.C6T:p.F2F,SETMAR:NM_006515:exon1:c.C6T:p.F2F,\t0\tNR_024022\tSETMAR\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr4\t1107150\t1107150\tC\tT\tchr4:1107125\t277.0\to\t603\t603\tTSS\texonic\tRNF212\tnonsynonymous SNV\tRNF212:NM_001193318:exon1:c.G103A:p.G35S,RNF212:NM_194439:exon1:c.G103A:p.G35S,RNF212:NM_001131034:exon1:c.G103A:p.G35S,\t0.037037\tNR_134676\tLOC105374344\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr5\t5140590\t5140590\tC\tT\tchr5:5139441\t398.5\to\t1500\t1500\tTSS\texonic\tADAMTS16\tnonsynonymous SNV\tADAMTS16:NM_139056:exon1:c.C10T:p.R4C,\t0\tNR_109915\tCTD-2297D10.2\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr6\t10695230\t10695230\tC\tT\tchr6:10694617\t-245.0\to\t735\t735\tTSS\texonic\tPAK1IP1\tsynonymous SNV\tPAK1IP1:NM_017906:exon1:c.C12T:p.V4V,\t0\tNR_026736\tC6orf52\tNR_TSS\t8.236e-06;1;121412\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr7\t2296538\t2296538\tC\tT\tchr7:2296353\t-263.5\to\t896\t896\torphan\texonic\tSNX8\tnonsynonymous SNV\tSNX8:NM_013321:exon10:c.G1255A:p.V419I,\t0.0909091\tNR_106895\tMIR6836\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr8\t6264193\t6264193\tC\tT\tchr8:6263582\t255.0\to\t711\t711\tTSS\texonic\tMCPH1\tnonsynonymous SNV\tMCPH1:NM_001172575:exon1:c.C5T:p.A2V,MCPH1:NM_001172574:exon1:c.C5T:p.A2V,MCPH1:NM_024596:exon1:c.C5T:p.A2V,\t0\tNR_040040\tLOC100287015\tNR_TSS\t1.662e-05;2;120358\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr9\t2622197\t2622197\tC\tT\tchr9:2621308\t-300.0\to\t2377\t2377\tTSS\texonic\tVLDLR\tnonsynonymous SNV\tVLDLR:NM_003383:exon1:c.C8T:p.T3M,VLDLR:NM_001018056:exon1:c.C8T:p.T3M,\t0\tNR_015375\tVLDLR-AS1\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr10\t8097629\t8097629\tC\tT\tchr10:8091374\t2777.0\to\t6955\t6955\tTSS\texonic\tGATA3\tnonsynonymous SNV\tGATA3:NM_001002295:exon2:c.C11T:p.T4M,GATA3:NM_002051:exon2:c.C11T:p.T4M,\t0\tNR_104330NR_104328\tGATA3-AS1GATA3-AS1\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr11\t1330696\t1330696\tC\tT\tchr11:1330390\t248.5\to\t1108\t1108\tTSS\texonic\tTOLLIP\tsynonymous SNV\tTOLLIP:NM_019009:exon1:c.G33A:p.P11P,\t0\tNR_029409\tTOLLIP-AS1\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr12\t2800140\t2800140\tC\tT\tchr12:2800139\t-461.0\to\t923\t923\torphan\texonic\tCACNA1C\tsynonymous SNV\tCACNA1C:NM_001129836:exon47:c.C6243T:p.A2081A,CACNA1C:NM_001129831:exon48:c.C6276T:p.A2092A,CACNA1C:NM_001129834:exon47:c.C6249T:p.A2083A,CACNA1C:NM_001129830:exon48:c.C6297T:p.A2099A,CACNA1C:NM_001129832:exon48:c.C6252T:p.A2084A,CACNA1C:NM_001129846:exon46:c.C6159T:p.A2053A,CACNA1C:NM_000719:exon47:c.C6192T:p.A2064A,CACNA1C:NM_001129829:exon47:c.C6315T:p.A2105A,CACNA1C:NM_001129843:exon47:c.C6192T:p.A2064A,CACNA1C:NM_001129835:exon47:c.C6249T:p.A2083A,CACNA1C:NM_001167623:exon47:c.C6192T:p.A2064A,CACNA1C:NM_001167624:exon48:c.C6297T:p.A2099A,CACNA1C:NM_199460:exon50:c.C6441T:p.A2147A,CACNA1C:NM_001129840:exon47:c.C6192T:p.A2064A,CACNA1C:NM_001129837:exon46:c.C6216T:p.A2072A,CACNA1C:NM_001129827:exon49:c.C6336T:p.A2112A,CACNA1C:NM_001129838:exon46:c.C6216T:p.A2072A,CACNA1C:NM_001129842:exon47:c.C6192T:p.A2064A,CACNA1C:NM_001129844:exon47:c.C6183T:p.A2061A,CACNA1C:NM_001129833:exon47:c.C6249T:p.A2083A,CACNA1C:NM_001167625:exon47:c.C6372T:p.A2124A,CACNA1C:NM_001129839:exon46:c.C6210T:p.A2070A,CACNA1C:NM_001129841:exon47:c.C6192T:p.A2064A,\t0.05\tNR_045725\tCACNA1C-AS1\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr13\t28494281\t28494281\tC\tT\tchr13:28494001\t-769.5\to\t2098\t2098\tTSS\texonic\tPDX1\tsynonymous SNV\tPDX1:NM_000209:exon1:c.C6T:p.N2N,\t0\tNR_047484\tPDX1-AS1\tNR_TSS\t2.458e-04;19;77302\t0.00055;2.0;3646.0\t0.0004;2.0;5008.0\t0.00046;4.0;8654.0\t0;0;0;0.002;0\t1;1\n",
      "chr14\t36277936\t36277936\tC\tT\tchr14:36277778\t263.5\to\t842\t842\tTSS\texonic\tRALGAPA1\tnonsynonymous SNV\tRALGAPA1:NM_001283043:exon1:c.G106A:p.E36K,RALGAPA1:NM_014990:exon1:c.G106A:p.E36K,RALGAPA1:NM_001283044:exon1:c.G106A:p.E36K,RALGAPA1:NM_194301:exon1:c.G106A:p.E36K,\t0\tNR_104269\tRALGAPA1P1\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr15\t35838331\t35838331\tC\tT\tchr15:35838126\t-10.0\to\t389\t389\tTSS\texonic\tDPH6\tsynonymous SNV\tDPH6:NM_001141972:exon1:c.G12A:p.A4A,DPH6:NM_080650:exon1:c.G12A:p.A4A,\t0\tNR_038251\tDPH6-AS1\tNR_TSS\t1.739e-05;1;57490\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr16\t2139804\t2139804\tC\tT\tchr16:2139803\t-1183.0\to\t2367\t2367\torphan\texonic\tPKD1\tnonsynonymous SNV\tPKD1:NM_001009944:exon46:c.G12836A:p.R4279Q,PKD1:NM_000296:exon46:c.G12833A:p.R4278Q,\t0.48\tNR_135175\tLOC105371049\tNR_TSS\t1.693e-05;2;118128\t.;.;.\t0.0002;1.0;5008.0\t0.0002;1.0;5008.0\t0;0.0014;0;0;0\t0;1\n",
      "chr17\t1619879\t1619879\tC\tT\tchr17:1619204\t245.5\to\t858\t858\tTSS\texonic\tWDR81\tnonsynonymous SNV\tWDR81:NM_001163673:exon1:c.C10T:p.R4C,\t0\tNR_028505NR_028504NR_028503NR_028502\tMIR22HGMIR22HGMIR22HGMIR22HG\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr18\t19751120\t19751120\tC\tT\tchr18:19744936\t2470.0\to\t7427\t7427\tTSS\texonic\tGATA6\tsynonymous SNV\tGATA6:NM_005257:exon2:c.C15T:p.D5D,\t0.0769231\tNR_102763\tGATA6-AS1\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr19\t804300\t804300\tC\tT\tchr19:804218\t-1141.0\to\t2445\t2445\torphan\texonic\tPTBP1\tsynonymous SNV\tPTBP1:NM_031991:exon5:c.C297T:p.I99I,PTBP1:NM_002819:exon5:c.C297T:p.I99I,PTBP1:NM_031990:exon5:c.C297T:p.I99I,\t0.933333\tNR_039900\tMIR4745\tNR_TSS\t8.237e-06;1;121410\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr20\t1373478\t1373478\tC\tT\tchr20:1373259\t110.5\to\t658\t658\tTSS\texonic\tFKBP1A\tnonsynonymous SNV\tFKBP1A:NM_054014:exon2:c.G85A:p.G29R,FKBP1A:NM_000801:exon2:c.G85A:p.G29R,FKBP1A:NM_001199786:exon2:c.G85A:p.D29N,\t0.0526316\tNR_106929\tMIR6869\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr21\t33984426\t33984426\tC\tT\tchr21:33984367\t483.0\to\t1083\t1083\tTSS\texonic\tC21orf59\tnonsynonymous SNV\tC21orf59:NM_021254:exon1:c.G128A:p.R43H,\t0.037037\tNR_036552\tC21orf59\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chr22\t19706268\t19706268\tC\tT\tchr22:19705510\t234.0\to\t1047\t1047\tTSS\texonic\tSEPT5\tnonsynonymous SNV\tSEPT5:NM_001009939:exon1:c.C8T:p.S3L,\t0\tNR_037612\tSEPT5-GP1BB\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "chrX\t38664214\t38664214\tC\tT\tchrX:38662906\t378.5\to\t1858\t1858\torphan\texonic\tMID1IP1\tsynonymous SNV\tMID1IP1:NM_021242:exon3:c.C15T:p.C5C,MID1IP1:NM_001098791:exon2:c.C15T:p.C5C,MID1IP1:NM_001098790:exon3:c.C15T:p.C5C,\t0\tNR_046706\tMID1IP1-AS1\tNR_TSS\t.;.;.\t.;.;.\t.;.;.\t.;.;.\t.;.;.;.;.\t0;0\n",
      "['noCGI_noexon', 'NR_TSS_noexon', 'noCGI_exon', 'NR_intragenic_noexon', 'intergenic_noexon', 'NM_TSS_noexon', 'NM_TSS_exon', 'NM_intragenic_noexon', 'NM_intragenic_exon', 'NR_TSS_exon', 'intergenic_exon']\n",
      "[3814370, 19742, 88994, 6302, 74206, 140628, 24733, 71073, 28756, 10, 78]\n",
      "[811413, 630, 16288, 548, 4342, 3905, 947, 8878, 3545, 4, 9]\n",
      "[665614, 528, 11457, 444, 3435, 2620, 601, 6470, 2478, 0, 6]\n",
      "[958723, 872, 19678, 715, 5661, 5032, 1249, 10730, 4468, 4, 11]\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# start analysis\n",
    "# Figure 1e\n",
    "list_class = []\n",
    "total_class = []\n",
    "thousand_class = []\n",
    "pcawg_class = []\n",
    "union_class = []\n",
    "mm=0\n",
    "for i in range(1,25):\n",
    "    input_file = file(\"revision_CpG_CtoT_PCAWG_chr%s.txt\" % str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\"))\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[11] == 'exonic':\n",
    "            input_exon = '_exon'\n",
    "        else:\n",
    "            input_exon = '_noexon'\n",
    "        \n",
    "        if input_split[7] == 'x':\n",
    "            input_class = 'noCGI' + input_exon\n",
    "        else:\n",
    "            input_class = input_split[18]\n",
    "            if input_class == 'both_intragenic':\n",
    "                input_class = 'NM_intragenic'\n",
    "            input_class = input_class + input_exon\n",
    "            if input_class == 'NR_TSS_exon':\n",
    "                print input_line\n",
    "                mm+=1\n",
    "                if mm>10:\n",
    "                    break\n",
    "\n",
    "        c_index = 99\n",
    "        if input_class not in list_class:\n",
    "            list_class.append(input_class)\n",
    "            total_class.append(1)\n",
    "            if input_split[20].split(';')[0] != '.':\n",
    "                pcawg_class.append(1)\n",
    "            else:\n",
    "                pcawg_class.append(0)\n",
    "\n",
    "            if input_split[21].split(';')[0] != '.': \n",
    "                thousand_class.append(1)\n",
    "            else:\n",
    "                thousand_class.append(0)\n",
    "\n",
    "            if input_split[22].split(';')[0] != '.':\n",
    "                union_class.append(1)\n",
    "            else:\n",
    "                union_class.append(0)\n",
    "\n",
    "        else:\n",
    "            c_index = list_class.index(input_class)\n",
    "            total_class[c_index] += 1\n",
    "\n",
    "            if input_split[20].split(';')[0] == '.':\n",
    "                'blank'\n",
    "            else:\n",
    "                pcawg_class[c_index] += 1\n",
    "            if input_split[21].split(';')[0] == '.':\n",
    "                'blank'\n",
    "            else:\n",
    "                thousand_class[c_index] += 1\n",
    "            if input_split[22].split(';')[0] == '.':\n",
    "                'blank'\n",
    "            else:\n",
    "                union_class[c_index] += 1\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "    \n",
    "print list_class\n",
    "print total_class\n",
    "print thousand_class\n",
    "print pcawg_class\n",
    "print union_class\n",
    "\n",
    "output_file = file(\"figure1e_exonic_nouse.txt\",\"w\")\n",
    "output_file.write('\\t'.join(list_class) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,total_class)) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,thousand_class)) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,pcawg_class)) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,union_class)))\n",
    "output_file.close()\n",
    "\n",
    "\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\t2\t3\n",
      "chr1=[]\n",
      "chr2=[]\n",
      "chr3=[]\n",
      "chr4=[]\n",
      "chr5=[]\n",
      "chr6=[]\n",
      "chr7=[]\n",
      "chr8=[]\n",
      "chr9=[]\n",
      "chr10=[]\n",
      "chr11=[]\n",
      "chr12=[]\n",
      "chr13=[]\n",
      "chr14=[]\n",
      "chr15=[]\n",
      "chr16=[]\n",
      "chr17=[]\n",
      "chr18=[]\n",
      "chr19=[]\n",
      "chr20=[]\n",
      "chr21=[]\n",
      "chr22=[]\n",
      "chrX=[]\n",
      "chrY=[]\n"
     ]
    }
   ],
   "source": [
    "a=[1,2,3]\n",
    "\n",
    "print '\\t'.join(map(str,a))\n",
    "\n",
    "\n",
    "for i in range(1,25):\n",
    "    print \"chr%s=[]\" %str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "info complete\n",
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#CpG re-classification\n",
    "#['TSS_coding','TSS_noncoding','Intergenic','Intragenic_coding','Intragenic_noncoding','noCGI','miscellaneous']\n",
    "import sys, os, gzip\n",
    "input_file = file(\"/home/users/jhyouk/89_backup_Workstation/jhyouk/CpG/cpgisland_classification_201608/cpgIslandExt_sorted.txt\")\n",
    "input_line = input_file.readline().strip()\n",
    "\n",
    "info_file = file(\"/home/users/jhyouk/89_backup_Workstation/jhyouk/CpG/cpgisland_classification_201608/refGene0801.txt\")\n",
    "info_line = info_file.readline().strip()\n",
    "\n",
    "output_file = file(\"CGI_classification_revised.txt\",\"w\")\n",
    "info_list = []\n",
    "for i in range(1,25):\n",
    "    info_list.append([])\n",
    "\n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    info_chr = info_split[2]\n",
    "    info_index = int(info_chr.replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\")) - 1\n",
    "    if info_split[3] == '+':\n",
    "        info_tss = int(info_split[4])+1\n",
    "    else:\n",
    "        info_tss = int(info_split[5])\n",
    "        \n",
    "    info_list[info_index].append([int(info_split[4])+1,int(info_split[5]), info_split[1][0:2], int(info_tss)])\n",
    "    info_line = info_file.readline().strip()\n",
    "print 'info complete'\n",
    "prev_chr = 'chr0'\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_cgi = input_split[1] + ':' + input_split[3]\n",
    "    \n",
    "    input_chr = input_split[1]\n",
    "    if prev_chr != input_chr:\n",
    "        print input_chr\n",
    "        prev_chr = input_chr\n",
    "        \n",
    "    input_index = int(input_chr.replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\")) - 1\n",
    "    input_start = int(input_split[3]) + 1\n",
    "    input_end = int(input_split[4])\n",
    "    \n",
    "    input_class = []\n",
    "    for [info_start,info_end,info_class,info_tss] in info_list[input_index]:\n",
    "        if (input_start -100 <= info_tss) and  (info_tss <= input_end + 100):\n",
    "            temp_class = info_class + '_TSS'\n",
    "            if temp_class not in input_class:\n",
    "                input_class.append(temp_class)\n",
    "        else:\n",
    "            'blank'\n",
    "            \n",
    "        if input_start > info_end or input_end < info_start:\n",
    "            'blank'\n",
    "        else:\n",
    "            temp_class = info_class + '_intragenic'\n",
    "            tt = info_class + '_TSS'\n",
    "            if temp_class not in input_class and (tt not in input_class):\n",
    "                input_class.append(temp_class)\n",
    "    if len(input_class) == 0:\n",
    "        input_class.append(\"intergenic\")\n",
    "\n",
    "    \n",
    "    if 'NM_TSS' in input_class and 'NM_intragenic' in input_class:\n",
    "        input_class.remove('NM_intragenic')\n",
    "    if 'NR_TSS' in input_class and 'NR_intragenic' in input_class:\n",
    "        input_class.remove('NR_intragenic')\n",
    "    output_file.write(input_line + '\\t' + ','.join(input_class) + '\\n')\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    \n",
    "\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3433\n",
      "['NR_TSS', 'NR_intragenic', 'intergenic', 'NM_TSS', 'NM_intragenic']\n",
      "[712, 628, 5696, 9567, 7682]\n"
     ]
    }
   ],
   "source": [
    "input_file = file(\"CGI_classification_revised.txt\")\n",
    "input_line = input_file.readline().strip()\n",
    "n=0\n",
    "list_class = []\n",
    "n_class = []\n",
    "while input_line:\n",
    "    input_split = input_line.split('\\t')\n",
    "    input_class = input_split[12].split(',')\n",
    "\n",
    "        \n",
    "    if len(input_class) >1:\n",
    "        n+=1\n",
    "    else:\n",
    "        if input_class[0] not in list_class:\n",
    "            list_class.append(input_class[0])\n",
    "            n_class.append(1)\n",
    "        else:\n",
    "            n_class[list_class.index(input_class[0])] +=1\n",
    "\n",
    "    input_line = input_file.readline().strip()\n",
    "print n\n",
    "print list_class\n",
    "print n_class\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "info_list has been completed\n",
      "chr1\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-76-10591585c78f>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     39\u001b[0m         \u001b[0minput_distance\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m999999999\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     40\u001b[0m         \u001b[0minput_decision\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 41\u001b[0;31m         \u001b[0;32mfor\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0minfo_CGI\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0minfo_start\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0minfo_end\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0minfo_size\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0minfo_class\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32min\u001b[0m \u001b[0minfo_list\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0minput_chr\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     42\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0minput_pos\u001b[0m \u001b[0;34m>=\u001b[0m \u001b[0minfo_start\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0minput_pos\u001b[0m \u001b[0;34m<=\u001b[0m \u001b[0minfo_end\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     43\u001b[0m                 \u001b[0minfo\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'o'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0minfo_CGI\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'0'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0minfo_size\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0minfo_class\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import sys, os\n",
    "\n",
    "ii = sys.argv[1]\n",
    "info_file = file(\"CGI_classification_revised.txt\")\n",
    "info_line = info_file.readline().strip()\n",
    "info_list = []\n",
    "for i in range(1,25):\n",
    "    info_list.append([])\n",
    "    \n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    info_chr = info_split[1]\n",
    "    info_index = int(info_chr.replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\")) - 1\n",
    "    info_start = int(info_split[3]) + 1\n",
    "    info_end = int(info_split[4])\n",
    "    info_CGI = info_chr + ':' + info_split[3]\n",
    "    \n",
    "    info_list[info_index].append([info_CGI,info_start,info_end,info_split[6],info_split[12]])\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "print 'info_list has been completed'\n",
    "info_file.close()\n",
    "\n",
    "for i in range(int(ii),int(ii)+1):\n",
    "    print 'chr' + str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\")\n",
    "    input_file = file(\"revision_CpG_CtoT_PCAWG_chr%s.txt\" %str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\"))\n",
    "    output_file = file(\"revision_CpG_CtoT_PCAWG_CGI_reclassification_chr%s.txt\" %str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\"),\"w\")\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    output_file.write(input_line+'\\tIn_CGI\\tnear_CGI\\tdistace_from_CGI\\tCGI_size\\tCGI_class\\n')\n",
    "    \n",
    "    input_line = input_file.readline().strip()\n",
    "    prev_CGI = ''\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        input_chr = int(input_split[0].replace(\"chr\",\"\").replace(\"X\",\"23\").replace(\"Y\",\"24\"))\n",
    "        input_pos = int(input_split[1])\n",
    "        info = ['','','','','']\n",
    "        \n",
    "        input_distance = 999999999\n",
    "        input_decision = ''\n",
    "        for [info_CGI,info_start,info_end,info_size,info_class] in info_list[input_chr-1]:\n",
    "            if input_pos >= info_start and input_pos <= info_end:\n",
    "                info = ['o',info_CGI,'0',info_size,info_class]                \n",
    "                break\n",
    "            else:\n",
    "                temp_distance1 = input_pos - info_start\n",
    "                temp_distance2 = input_pos - info_end\n",
    "                if min(abs(temp_distance1),abs(temp_distance2)) == abs(temp_distance1):\n",
    "                    temp_distance = temp_distance1\n",
    "                else:\n",
    "                    temp_distance = temp_distance2\n",
    "                \n",
    "                if input_distance > abs(temp_distance):\n",
    "                    input_distance = abs(temp_distance)\n",
    "                    info = ['x',info_CGI,str(temp_distance),info_size,info_class]\n",
    "        \n",
    "        output_file.write(input_line+'\\t' + '\\t'.join(info) + '\\n')\n",
    "        input_line = input_file.readline().strip()\n",
    "        \n",
    "    \n",
    "    \n",
    "    input_file.close()\n",
    "    output_file.close()\n",
    "\n",
    "print 'THE END'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python 03_reclassfication_add_individual_genotype.py 1\n",
      "python 03_reclassfication_add_individual_genotype.py 2\n",
      "python 03_reclassfication_add_individual_genotype.py 3\n",
      "python 03_reclassfication_add_individual_genotype.py 4\n",
      "python 03_reclassfication_add_individual_genotype.py 5\n",
      "python 03_reclassfication_add_individual_genotype.py 6\n",
      "python 03_reclassfication_add_individual_genotype.py 7\n",
      "python 03_reclassfication_add_individual_genotype.py 8\n",
      "python 03_reclassfication_add_individual_genotype.py 9\n",
      "python 03_reclassfication_add_individual_genotype.py 10\n",
      "python 03_reclassfication_add_individual_genotype.py 11\n",
      "python 03_reclassfication_add_individual_genotype.py 12\n",
      "python 03_reclassfication_add_individual_genotype.py 13\n",
      "python 03_reclassfication_add_individual_genotype.py 14\n",
      "python 03_reclassfication_add_individual_genotype.py 15\n",
      "python 03_reclassfication_add_individual_genotype.py 16\n",
      "python 03_reclassfication_add_individual_genotype.py 17\n",
      "python 03_reclassfication_add_individual_genotype.py 18\n",
      "python 03_reclassfication_add_individual_genotype.py 19\n",
      "python 03_reclassfication_add_individual_genotype.py 20\n",
      "python 03_reclassfication_add_individual_genotype.py 21\n",
      "python 03_reclassfication_add_individual_genotype.py 22\n",
      "python 03_reclassfication_add_individual_genotype.py 23\n",
      "python 03_reclassfication_add_individual_genotype.py 24\n"
     ]
    }
   ],
   "source": [
    "for i in range(1,25):\n",
    "    print 'python 03_reclassfication_add_individual_genotype.py %s' %i\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['noCGINoE', 'NR_TSSNoE', 'noCGIsyn', 'noCGInon', 'NR_intragenicNoE', 'intergenicNoE', 'NM_TSSNoE', 'NM_TSSnon', 'NM_TSSsyn', 'NM_TSSsto', 'noCGIsto', 'NM_intragenicNoE', 'NM_intragenicsyn', 'NM_intragenicnon', 'NM_intragenicsto', 'noCGIunk', 'miscellaneousNoE', 'miscellaneoussyn', 'miscellaneousnon', 'miscellaneoussto', 'NM_intragenicunk', 'NM_TSSunk', 'miscellaneousunk']\n",
      "[50557388, 117200, 448684, 936645, 73452, 601434, 1680847, 181835, 115450, 2784, 62091, 442245, 114587, 194250, 2840, 16830, 503538, 40531, 65077, 994, 5094, 3680, 1640]\n",
      "[12010891, 3594, 107078, 148688, 3357, 28699, 44640, 4474, 4103, 31, 3777, 38554, 13188, 14221, 85, 3190, 17091, 2411, 2730, 21, 423, 87, 70]\n",
      "[9728556, 2706, 82676, 99081, 2561, 22071, 30334, 2853, 4121, 26, 2578, 28054, 11457, 9333, 51, 2373, 12110, 2248, 1782, 15, 307, 80, 46]\n",
      "[14027551, 4797, 131752, 179052, 4402, 37422, 58863, 5990, 6700, 46, 4748, 47845, 18187, 18078, 110, 3870, 22421, 3562, 3559, 27, 539, 128, 90]\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# Figure 1e using CGI reclassfication\n",
    "list_class = []\n",
    "total_class = []\n",
    "thousand_class = []\n",
    "pcawg_class = []\n",
    "union_class = []\n",
    "mm=0\n",
    "\n",
    "for i in range(1,25):\n",
    "    input_file = file(\"revision_CpG_CtoT_PCAWG_CGI_reclassification_chr%s.txt\" % str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\"))\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        input_exon = input_split[13][0:3]\n",
    "        \n",
    "        input_exon = ''\n",
    "        if input_split[25] == 'x':\n",
    "            input_class = 'noCGI' + input_exon\n",
    "        else:\n",
    "            input_class = input_split[29]\n",
    "            if len(input_class.split(',')) > 1:\n",
    "                input_class = 'miscellaneous'\n",
    "            input_class = input_class + input_exon\n",
    "\n",
    "        c_index = 99\n",
    "        \n",
    "        if input_class not in list_class:\n",
    "            list_class.append(input_class)\n",
    "            total_class.append(1)\n",
    "            if input_split[20].split(';')[0] != '.':\n",
    "                pcawg_class.append(1)\n",
    "            else:\n",
    "                pcawg_class.append(0)\n",
    "\n",
    "            if input_split[21].split(';')[0] != '.': \n",
    "                thousand_class.append(1)\n",
    "            else:\n",
    "                thousand_class.append(0)\n",
    "\n",
    "            if input_split[22].split(';')[0] != '.':\n",
    "                union_class.append(1)\n",
    "            else:\n",
    "                union_class.append(0)\n",
    "\n",
    "        else:\n",
    "            c_index = list_class.index(input_class)\n",
    "            total_class[c_index] += 1\n",
    "\n",
    "            if input_split[20].split(';')[0] == '.':\n",
    "                'blank'\n",
    "            else:\n",
    "                pcawg_class[c_index] += 1\n",
    "            if input_split[21].split(';')[0] == '.':\n",
    "                'blank'\n",
    "            else:\n",
    "                thousand_class[c_index] += 1\n",
    "            if input_split[22].split(';')[0] == '.':\n",
    "                'blank'\n",
    "            else:\n",
    "                union_class[c_index] += 1\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "    \n",
    "print list_class\n",
    "print total_class\n",
    "print thousand_class\n",
    "print pcawg_class\n",
    "print union_class\n",
    "\n",
    "output_file = file(\"figure1e_reclassfication_selectivepressure.txt\",\"w\")\n",
    "output_file.write('\\t'.join(list_class) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,total_class)) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,thousand_class)) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,pcawg_class)) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,union_class)))\n",
    "output_file.close()\n",
    "\n",
    "\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['NoExon', 'synonymous SNV', 'nonsynonymous SNV', 'stopgain', 'unknown']\n"
     ]
    }
   ],
   "source": [
    "list_syn = []\n",
    "for i in range(1,2):\n",
    "    input_file = file(\"revision_CpG_CtoT_PCAWG_CGI_reclassification_chr%s.txt\" % str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\"))\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        if input_split[13] not in list_syn:\n",
    "            list_syn.append(input_split[13])\n",
    "            \n",
    "        input_line = input_file.readline().strip()\n",
    "print list_syn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['noCGI', 'NR_TSS', 'NR_intragenic', 'intergenic', 'NM_TSS', 'NM_intragenic', 'miscellaneous']\n",
      "[52021638, 117200, 73452, 601434, 1984596, 759016, 611780]\n",
      "[3702980, 979, 949, 8280, 13210, 18269, 5659]\n",
      "[4267756, 1220, 1136, 9760, 16264, 19245, 6474]\n",
      "[6467174, 1837, 1722, 14941, 25504, 31003, 10554]\n",
      "[3677617, 1033, 944, 8491, 14292, 17837, 5836]\n",
      "[4203270, 1172, 1138, 9777, 16666, 20853, 7003]\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# Figure 1e using 1000 genomes race information\n",
    "list_class = []\n",
    "total_class = []\n",
    "\n",
    "eas_class = []\n",
    "amr_class = []\n",
    "afr_class = []\n",
    "eur_class = []\n",
    "sas_class = []\n",
    "\n",
    "\n",
    "for i in range(1,25):\n",
    "    input_file = file(\"revision_CpG_CtoT_PCAWG_CGI_reclassification_chr%s.txt\" % str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\"))\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        input_exon = ''\n",
    "        \n",
    "        \n",
    "        if input_split[25] == 'x':\n",
    "            input_class = 'noCGI' + input_exon\n",
    "        else:\n",
    "            input_class = input_split[29]\n",
    "            if len(input_class.split(',')) > 1:\n",
    "                input_class = 'miscellaneous'\n",
    "            input_class = input_class + input_exon\n",
    "\n",
    "        c_index = 99\n",
    "        \n",
    "        if input_class not in list_class:\n",
    "            list_class.append(input_class)\n",
    "            total_class.append(1)\n",
    "            temp = [0,0,0,0,0]\n",
    "            for j in range(0,5):\n",
    "                if input_split[23].split(';')[j] == '.':\n",
    "                    temp[j] = 0.0\n",
    "                else:\n",
    "                    temp[j] = float(input_split[23].split(';')[j])\n",
    "                \n",
    "            \n",
    "            if temp[0] > 0.0:\n",
    "                eas_class.append(1)\n",
    "            else:\n",
    "                eas_class.append(0)\n",
    "\n",
    "            if temp[1] > 0.0: \n",
    "                amr_class.append(1)\n",
    "            else:\n",
    "                amr_class.append(0)\n",
    "\n",
    "            if temp[2] > 0.0: \n",
    "                afr_class.append(1)\n",
    "            else:\n",
    "                afr_class.append(0)\n",
    "            if temp[3] > 0.0: \n",
    "                eur_class.append(1)\n",
    "            else:\n",
    "                eur_class.append(0)\n",
    "            if temp[4] > 0.0: \n",
    "                sas_class.append(1)\n",
    "            else:\n",
    "                sas_class.append(0)                \n",
    "\n",
    "        else:\n",
    "            temp = [0,0,0,0,0]\n",
    "            for j in range(0,5):\n",
    "                if input_split[23].split(';')[j] == '.':\n",
    "                    temp[j] = 0.0\n",
    "                else:\n",
    "                    temp[j] = float(input_split[23].split(';')[j])\n",
    "                    \n",
    "            c_index = list_class.index(input_class)\n",
    "            total_class[c_index] += 1\n",
    "\n",
    "            if temp[0] == 0:\n",
    "                'blank'\n",
    "            else:\n",
    "                eas_class[c_index] += 1\n",
    "                \n",
    "            if temp[1] == 0:\n",
    "                'blank'\n",
    "            else:\n",
    "                amr_class[c_index] += 1\n",
    "                \n",
    "            if temp[2] == 0:\n",
    "                'blank'\n",
    "            else:\n",
    "                afr_class[c_index] += 1\n",
    "                \n",
    "            if temp[3] == 0:\n",
    "                'blank'\n",
    "            else:\n",
    "                eur_class[c_index] += 1\n",
    "                \n",
    "            if temp[4] == 0:\n",
    "                'blank'\n",
    "            else:\n",
    "                sas_class[c_index] += 1                \n",
    "                \n",
    "        input_line = input_file.readline().strip()\n",
    "    \n",
    "print list_class\n",
    "print total_class\n",
    "print eas_class\n",
    "print amr_class\n",
    "print afr_class\n",
    "print eur_class\n",
    "print sas_class\n",
    "\n",
    "output_file = file(\"figure1e_race.txt\",\"w\")\n",
    "output_file.write('\\t'.join(list_class) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,total_class)) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,eas_class)) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,amr_class)) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,afr_class)) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,eur_class)) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,sas_class)))\n",
    "output_file.close()\n",
    "\n",
    "\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "['noCGI', 'NR_TSS', 'NR_intragenic', 'intergenic', 'NM_TSS', 'NM_intragenic', 'miscellaneous']\n",
      "[51703250, 116304, 73452, 597266, 1982770, 758164, 611056]\n",
      "[[232420, 96, 91, 739, 960, 902, 367], [232843, 93, 105, 679, 931, 896, 361], [232621, 103, 84, 713, 941, 860, 359]]\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "# analysis for individual variation (Except Y chromosome)\n",
    "\n",
    "list_class = ['noCGI', 'NR_TSS', 'NR_intragenic', 'intergenic', 'NM_TSS', 'NM_intragenic', 'miscellaneous']\n",
    "total_class = [0,0,0,0,0,0,0]\n",
    "\n",
    "individual_class = []\n",
    "for k in range(30,1853):\n",
    "    individual_class.append([0,0,0,0,0,0,0])\n",
    "    \n",
    "for i in range(1,24):\n",
    "    print 'chr' + str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\")\n",
    "    input_file = file(\"revision_CpG_CtoT_PCAWG_CGI_reclassification_chr%s_individual.txt\" % str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\"))\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        input_gt_pcawg = input_split[24].split(';')[0]\n",
    "        input_gt_thousand = input_split[24].split(';')[1]\n",
    "        \n",
    "        if input_split[25] == 'x':\n",
    "            input_class = 'noCGI'\n",
    "        else:\n",
    "            input_class = input_split[29]\n",
    "            if len(input_class.split(',')) > 1:\n",
    "                input_class = 'miscellaneous' \n",
    "                         \n",
    "        c_index = list_class.index(input_class)\n",
    "        \n",
    "        total_class[c_index]+=1\n",
    "        for j in range(30,1853): #PCAWG\n",
    "            if input_split[j] == '.':\n",
    "                'blank'\n",
    "            else:\n",
    "                for kk in input_split[j].split('|'):\n",
    "                    if kk == input_gt_pcawg:\n",
    "                        individual_class[j-30][c_index] += 1\n",
    "                        break\n",
    "                    \n",
    "            \n",
    "        #for j in range(1853,4357): #Thousand\n",
    "            #if input_split[j] == '.':\n",
    "            #    'blank'\n",
    "            #else:            \n",
    "            #    for kk in input_split[j].split('|'):\n",
    "            #        if kk == input_gt_thousand:\n",
    "            #            individual_class[j-30][c_index] += 1\n",
    "            #            break\n",
    "\n",
    "                \n",
    "        input_line = input_file.readline().strip()\n",
    "    \n",
    "print list_class\n",
    "print total_class\n",
    "print individual_class[0:3]\n",
    "\n",
    "\n",
    "output_file = file(\"figure1e_individual_pcawg.txt\",\"w\")\n",
    "output_file.write('\\t'.join(list_class) + '\\n')\n",
    "output_file.write('\\t'.join(map(str,total_class)) + '\\n')\n",
    "for k in range(30,1853):\n",
    "    output_file.write('\\t'.join(map(str,individual_class[k-30])) + '\\n')\n",
    "\n",
    "output_file.close()\n",
    "\n",
    "print 'THE END'    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#Figure 2A,B\n",
    "import sys,os\n",
    "\n",
    "total=[]\n",
    "all_CGI=[]\n",
    "NM_TSS=[]\n",
    "NR_TSS=[]\n",
    "Intergenic=[]\n",
    "NM_intragenic=[]\n",
    "NR_intragenic=[]\n",
    "miscellaneous=[]\n",
    "\n",
    "NM_TSS_all=[]\n",
    "NR_TSS_all=[]\n",
    "Intergenic_all=[]\n",
    "NM_intragenic_all=[]\n",
    "NR_intragenic_all=[]\n",
    "miscellaneous_all=[]\n",
    "\n",
    "for i in range(0,10002):\n",
    "    total.append(0)\n",
    "    all_CGI.append(0)\n",
    "    NM_TSS.append(0)\n",
    "    NR_TSS.append(0)\n",
    "    Intergenic.append(0)\n",
    "    NM_intragenic.append(0)\n",
    "    NR_intragenic.append(0)\n",
    "    miscellaneous.append(0)\n",
    "    NM_TSS_all.append(0)\n",
    "    NR_TSS_all.append(0)\n",
    "    Intergenic_all.append(0)\n",
    "    NM_intragenic_all.append(0)\n",
    "    NR_intragenic_all.append(0)\n",
    "    miscellaneous_all.append(0)\n",
    "\n",
    "output_file = file(\"distance_by_group_Y.txt\",\"w\")\n",
    "    \n",
    "for i in range(1,25):\n",
    "    print 'chr' + str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\")\n",
    "    input_file = file(\"revision_CpG_CtoT_PCAWG_CGI_reclassification_chr%s.txt\" % str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\"))\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        total_af = input_split[22].split(';')[0]\n",
    "        distance = int(input_split[27])\n",
    "        if distance < 0:\n",
    "            input_index = 5000+distance\n",
    "        elif distance > 0:\n",
    "            input_index = 5000+distance\n",
    "            \n",
    "        input_class = input_split[29]\n",
    "        if len(input_class.split(',')) > 1:\n",
    "            input_class = 'miscellaneous'\n",
    "\n",
    "        if abs(distance) > 5000 or distance == 0:\n",
    "            input_line = input_file.readline().strip()\n",
    "            continue\n",
    "        else:\n",
    "            total[input_index]+=1\n",
    "            \n",
    "            if input_class == 'NM_TSS':\n",
    "                NM_TSS_all[input_index]+=1\n",
    "            elif input_class == 'NR_TSS':\n",
    "                NR_TSS_all[input_index]+=1\n",
    "            elif input_class == 'intergenic':\n",
    "                Intergenic_all[input_index]+=1\n",
    "            elif input_class == 'NM_intragenic':\n",
    "                NM_intragenic_all[input_index]+=1\n",
    "            elif input_class == 'NR_intragenic':\n",
    "                NR_intragenic_all[input_index]+=1\n",
    "            elif input_class == 'miscellaneous':\n",
    "                miscellaneous_all[input_index]+=1\n",
    "            else:\n",
    "                print 'Unexpected result'\n",
    "                print input_line\n",
    "                sys.exit(1)            \n",
    "                \n",
    "            if total_af == '.':\n",
    "                'blank'\n",
    "            else:\n",
    "                all_CGI[input_index]+=1\n",
    "                if input_class == 'NM_TSS':\n",
    "                    NM_TSS[input_index]+=1\n",
    "                elif input_class == 'NR_TSS':\n",
    "                    NR_TSS[input_index]+=1\n",
    "                elif input_class == 'intergenic':\n",
    "                    Intergenic[input_index]+=1\n",
    "                elif input_class == 'NM_intragenic':\n",
    "                    NM_intragenic[input_index]+=1\n",
    "                elif input_class == 'NR_intragenic':\n",
    "                    NR_intragenic[input_index]+=1\n",
    "                elif input_class == 'miscellaneous':\n",
    "                    miscellaneous[input_index]+=1\n",
    "                else:\n",
    "                    print 'Unexpected result'\n",
    "                    print input_line\n",
    "                    sys.exit(1)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "        \n",
    "\n",
    "for i in range(0,10002):\n",
    "    if i<5000:\n",
    "        j=i-5500\n",
    "    elif i>5000:\n",
    "        j=i-4500\n",
    "        \n",
    "    output_file.write(str(j)+'\\t' + str(total[i]) +'\\t' + str(all_CGI[i])+ '\\t' +  str(NM_TSS_all[i]) + '\\t' + str(NM_TSS[i]) + '\\t' + str(NR_TSS_all[i])+ '\\t' +   str(NR_TSS[i])+ '\\t' + str(Intergenic_all[i])+ '\\t' +  str(Intergenic[i])+ '\\t' + str(NM_intragenic_all[i])+ '\\t' +  str(NM_intragenic[i])+ '\\t' +  str(NR_intragenic_all[i])+ '\\t' +  str(NR_intragenic[i])+ '\\t' + str(miscellaneous_all[i]) + '\\t' +str(miscellaneous[i]) + '\\n')\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#Fig2c\n",
    "import sys,os\n",
    "\n",
    "total=[]\n",
    "all_CGI=[]\n",
    "NM_TSS=[]\n",
    "NR_TSS=[]\n",
    "Intergenic=[]\n",
    "NM_intragenic=[]\n",
    "NR_intragenic=[]\n",
    "miscellaneous=[]\n",
    "\n",
    "NM_TSS_all=[]\n",
    "NR_TSS_all=[]\n",
    "Intergenic_all=[]\n",
    "NM_intragenic_all=[]\n",
    "NR_intragenic_all=[]\n",
    "miscellaneous_all=[]\n",
    "\n",
    "for i in range(0,10002):\n",
    "    total.append(0)\n",
    "    all_CGI.append(0)\n",
    "    NM_TSS.append(0)\n",
    "    NR_TSS.append(0)\n",
    "    Intergenic.append(0)\n",
    "    NM_intragenic.append(0)\n",
    "    NR_intragenic.append(0)\n",
    "    miscellaneous.append(0)\n",
    "    NM_TSS_all.append(0)\n",
    "    NR_TSS_all.append(0)\n",
    "    Intergenic_all.append(0)\n",
    "    NM_intragenic_all.append(0)\n",
    "    NR_intragenic_all.append(0)\n",
    "    miscellaneous_all.append(0)\n",
    "\n",
    "output_file = file(\"methylation_by_distance_Y.txt\",\"w\")\n",
    "    \n",
    "for i in range(1,25):\n",
    "    print 'chr' + str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\")\n",
    "    input_file = file(\"revision_CpG_CtoT_PCAWG_CGI_reclassification_chr%s.txt\" % str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\"))\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        sperm_methyl = float(input_split[15]) #sper methylation information\n",
    "        \n",
    "        distance = int(input_split[27])\n",
    "        if distance < 0:\n",
    "            input_index = 5000+distance\n",
    "        elif distance > 0:\n",
    "            input_index = 5000+distance\n",
    "            \n",
    "        input_class = input_split[29]\n",
    "        if len(input_class.split(',')) > 1:\n",
    "            input_class = 'miscellaneous'\n",
    "\n",
    "        if abs(distance) > 5000 or distance == 0:\n",
    "            input_line = input_file.readline().strip()\n",
    "            continue\n",
    "        else:\n",
    "            total[input_index]+=1\n",
    "            \n",
    "            if input_class == 'NM_TSS':\n",
    "                NM_TSS_all[input_index]+=1\n",
    "            elif input_class == 'NR_TSS':\n",
    "                NR_TSS_all[input_index]+=1\n",
    "            elif input_class == 'intergenic':\n",
    "                Intergenic_all[input_index]+=1\n",
    "            elif input_class == 'NM_intragenic':\n",
    "                NM_intragenic_all[input_index]+=1\n",
    "            elif input_class == 'NR_intragenic':\n",
    "                NR_intragenic_all[input_index]+=1\n",
    "            elif input_class == 'miscellaneous':\n",
    "                miscellaneous_all[input_index]+=1\n",
    "            else:\n",
    "                print 'Unexpected result'\n",
    "                print input_line\n",
    "                sys.exit(1)            \n",
    "            \n",
    "            if input_class == 'NM_TSS':\n",
    "                NM_TSS[input_index]+=sperm_methyl\n",
    "            elif input_class == 'NR_TSS':\n",
    "                NR_TSS[input_index]+=sperm_methyl\n",
    "            elif input_class == 'intergenic':\n",
    "                Intergenic[input_index]+=sperm_methyl\n",
    "            elif input_class == 'NM_intragenic':\n",
    "                NM_intragenic[input_index]+=sperm_methyl\n",
    "            elif input_class == 'NR_intragenic':\n",
    "                NR_intragenic[input_index]+=sperm_methyl\n",
    "            elif input_class == 'miscellaneous':\n",
    "                miscellaneous[input_index]+=sperm_methyl\n",
    "            else:\n",
    "                print 'Unexpected result'\n",
    "                print input_line\n",
    "                sys.exit(1)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "        \n",
    "\n",
    "for i in range(0,10002):\n",
    "    if i<5000:\n",
    "        j=i-5500\n",
    "    elif i>5000:\n",
    "        j=i-4500\n",
    "        \n",
    "    output_file.write(str(j)+'\\t' + str(total[i]) +'\\t' + str(all_CGI[i])+ '\\t' +  str(NM_TSS_all[i]) + '\\t' + str(NM_TSS[i]) + '\\t' + str(NR_TSS_all[i])+ '\\t' +   str(NR_TSS[i])+ '\\t' + str(Intergenic_all[i])+ '\\t' +  str(Intergenic[i])+ '\\t' + str(NM_intragenic_all[i])+ '\\t' +  str(NM_intragenic[i])+ '\\t' +  str(NR_intragenic_all[i])+ '\\t' +  str(NR_intragenic[i])+ '\\t' + str(miscellaneous_all[i]) + '\\t' +str(miscellaneous[i]) + '\\n')\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#Figure 2D,E,F\n",
    "import sys,os,numpy\n",
    "\n",
    "\n",
    "CGI=[]\n",
    "CGI_info = []\n",
    "\n",
    "output_file = file(\"methylation_by_CGI.txt\",\"w\")\n",
    "output_file.write(\"CGI_name\\tinput_class\\ttotal\\tmutated\\tsize\\tmethylation\\n\")\n",
    "for i in range(1,25):\n",
    "    print 'chr' + str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\")\n",
    "    input_file = file(\"revision_CpG_CtoT_PCAWG_CGI_reclassification_chr%s.txt\" % str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\"))\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        distance = int(input_split[27])\n",
    "        if distance != 0:\n",
    "            input_line = input_file.readline().strip()\n",
    "            continue\n",
    "        else:\n",
    "            input_class = input_split[29]\n",
    "            cgi_name = input_split[26]\n",
    "            if len(input_class.split(',')) > 1:\n",
    "                input_class = 'miscellaneous'  \n",
    "                \n",
    "            if cgi_name not in CGI:\n",
    "                CGI.append(cgi_name)\n",
    "                #input_class, total, mutated, size, methylation\n",
    "                CGI_info.append([input_class,0,0,input_split[28],[]])\n",
    "            \n",
    "            cgi_index=CGI.index(cgi_name)\n",
    "            CGI_info[cgi_index][1] +=1\n",
    "            if input_split[22].split(';')[0] != '.':\n",
    "                CGI_info[cgi_index][2] +=1\n",
    "            \n",
    "            CGI_info[cgi_index][4].append(float(input_split[15]))\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "#print CGI\n",
    "#print CGI_info\n",
    "for j in range(0,len(CGI)):\n",
    "    output_file.write(CGI[j]+'\\t' + str(CGI_info[j][0]) +'\\t' + str(CGI_info[j][1])+ '\\t' +  str(CGI_info[j][2]) + '\\t' + str(CGI_info[j][3]) + '\\t' + str(numpy.mean(CGI_info[j][4])) + '\\n')\n",
    "output_file.close()\n",
    "print 'THE END'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#Figure3a\n",
    "import sys,os,numpy\n",
    "\n",
    "TSS_syn=[0,0,0,0]\n",
    "TSS_non=[0,0,0,0]\n",
    "TSS_sto=[0,0,0,0]\n",
    "intra_syn=[0,0,0,0]\n",
    "intra_non=[0,0,0,0]\n",
    "intra_sto=[0,0,0,0]\n",
    "noCGI_syn=[0,0,0,0]\n",
    "noCGI_non=[0,0,0,0]\n",
    "noCGI_sto=[0,0,0,0]\n",
    "\n",
    "output_file = file(\"fig3a_pie_nomethyl.txt\",\"w\")\n",
    "output_file.write(\"yesWGS\\tnoWGS\\tyesExAC\\tnoExAC\\n\")\n",
    "\n",
    "for i in range(1,25):\n",
    "    print 'chr' + str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\")\n",
    "    input_file = file(\"revision_CpG_CtoT_PCAWG_CGI_reclassification_chr%s.txt\" % str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\"))\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        input_methyl = float(input_split[15])\n",
    "        input_type = input_split[13]\n",
    "        input_distance = int(input_split[27])\n",
    "        #if input_methyl < 0.67:\n",
    "        #    input_line = input_file.readline().strip()\n",
    "        #    continue\n",
    "        if input_split[11] != 'exonic':\n",
    "            input_line = input_file.readline().strip()\n",
    "            continue \n",
    "            \n",
    "\n",
    "        input_class = input_split[29]\n",
    "        if len(input_class.split(',')) > 1:\n",
    "            input_class = 'miscellaneous'  \n",
    "\n",
    "        if input_type[0:3] == 'syn':\n",
    "            if input_distance == 0:\n",
    "                if input_class == 'NM_TSS' or input_class == 'NR_TSS':\n",
    "                    if input_split[22].split(';')[0] == '.': #WGS\n",
    "                        TSS_syn[1] += 1\n",
    "                    else:\n",
    "                        TSS_syn[0] += 1                            \n",
    "                    if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                        TSS_syn[3] += 1\n",
    "                    else:\n",
    "                        TSS_syn[2] += 1                            \n",
    "                elif input_class == 'NM_intragenic' or input_class == 'NR_intragenic':\n",
    "                    if input_split[22].split(';')[0] == '.': #WGS\n",
    "                        intra_syn[1] += 1\n",
    "                    else:\n",
    "                        intra_syn[0] += 1                            \n",
    "                    if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                        intra_syn[3] += 1\n",
    "                    else:\n",
    "                        intra_syn[2] += 1    \n",
    "\n",
    "            else: \n",
    "                if input_split[22].split(';')[0] == '.': #WGS\n",
    "                    noCGI_syn[1] += 1\n",
    "                else:\n",
    "                    noCGI_syn[0] += 1                            \n",
    "                if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                    noCGI_syn[3] += 1\n",
    "                else:\n",
    "                    noCGI_syn[2] += 1     \n",
    "\n",
    "        elif input_type[0:3] == 'non':\n",
    "            if input_distance == 0:\n",
    "                if input_class == 'NM_TSS' or input_class == 'NR_TSS':\n",
    "                    if input_split[22].split(';')[0] == '.': #WGS\n",
    "                        TSS_non[1] += 1\n",
    "                    else:\n",
    "                        TSS_non[0] += 1                            \n",
    "                    if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                        TSS_non[3] += 1\n",
    "                    else:\n",
    "                        TSS_non[2] += 1                            \n",
    "                elif input_class == 'NM_intragenic' or input_class == 'NR_intragenic':\n",
    "                    if input_split[22].split(';')[0] == '.': #WGS\n",
    "                        intra_non[1] += 1\n",
    "                    else:\n",
    "                        intra_non[0] += 1                            \n",
    "                    if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                        intra_non[3] += 1\n",
    "                    else:\n",
    "                        intra_non[2] += 1    \n",
    "\n",
    "            else: \n",
    "\n",
    "                if input_split[22].split(';')[0] == '.': #WGS\n",
    "                    noCGI_non[1] += 1\n",
    "                else:\n",
    "                    noCGI_non[0] += 1                            \n",
    "                if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                    noCGI_non[3] += 1\n",
    "                else:\n",
    "                    noCGI_non[2] += 1    \n",
    "\n",
    "\n",
    "        elif input_type[0:3] == 'sto':\n",
    "            if input_distance == 0:\n",
    "                if input_class == 'NM_TSS' or input_class == 'NR_TSS':\n",
    "                    if input_split[22].split(';')[0] == '.': #WGS\n",
    "                        TSS_sto[1] += 1\n",
    "                    else:\n",
    "                        TSS_sto[0] += 1                            \n",
    "                    if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                        TSS_sto[3] += 1\n",
    "                    else:\n",
    "                        TSS_sto[2] += 1                            \n",
    "                elif input_class == 'NM_intragenic' or input_class == 'NR_intragenic':\n",
    "                    if input_split[22].split(';')[0] == '.': #WGS\n",
    "                        intra_sto[1] += 1\n",
    "                    else:\n",
    "                        intra_sto[0] += 1                            \n",
    "                    if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                        intra_sto[3] += 1\n",
    "                    else:\n",
    "                        intra_sto[2] += 1    \n",
    "\n",
    "            else: \n",
    "\n",
    "                if input_split[22].split(';')[0] == '.': #WGS\n",
    "                    noCGI_sto[1] += 1\n",
    "                else:\n",
    "                    noCGI_sto[0] += 1                            \n",
    "                if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                    noCGI_sto[3] += 1\n",
    "                else:\n",
    "                    noCGI_sto[2] += 1                   \n",
    "        elif input_type[0:3] == 'unk':\n",
    "            'blank'\n",
    "        else:\n",
    "            print input_line\n",
    "            sys.exit(1)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.write('\\t'.join(map(str,TSS_syn))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,TSS_non))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,TSS_sto))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,intra_syn))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,intra_non))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,intra_sto))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,noCGI_syn))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,noCGI_non))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,noCGI_sto)))\n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "\n",
    "#join(map(str,total_class)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\n",
      "chr2\n",
      "chr3\n",
      "chr4\n",
      "chr5\n",
      "chr6\n",
      "chr7\n",
      "chr8\n",
      "chr9\n",
      "chr10\n",
      "chr11\n",
      "chr12\n",
      "chr13\n",
      "chr14\n",
      "chr15\n",
      "chr16\n",
      "chr17\n",
      "chr18\n",
      "chr19\n",
      "chr20\n",
      "chr21\n",
      "chr22\n",
      "chrX\n",
      "chrY\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#Figure 3b\n",
    "import sys,os,numpy\n",
    "\n",
    "TSS_syn=[0,0,0,0,0,0]\n",
    "TSS_non=[0,0,0,0,0,0]\n",
    "TSS_sto=[0,0,0,0,0,0]\n",
    "intra_syn=[0,0,0,0,0,0]\n",
    "intra_non=[0,0,0,0,0,0]\n",
    "intra_sto=[0,0,0,0,0,0]\n",
    "noCGI_syn=[0,0,0,0,0,0]\n",
    "noCGI_non=[0,0,0,0,0,0]\n",
    "noCGI_sto=[0,0,0,0,0,0]\n",
    "\n",
    "output_file = file(\"fig3b.txt\",\"w\")\n",
    "\n",
    "for i in range(1,25):\n",
    "    print 'chr' + str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\")\n",
    "    input_file = file(\"revision_CpG_CtoT_PCAWG_CGI_reclassification_chr%s.txt\" % str(i).replace(\"23\",\"X\").replace(\"24\",\"Y\"))\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        input_methyl = float(input_split[15])\n",
    "        input_type = input_split[13]\n",
    "        input_distance = int(input_split[27])\n",
    "        if input_methyl < 0.67:\n",
    "            input_line = input_file.readline().strip()\n",
    "            continue\n",
    "        if input_split[11] != 'exonic':\n",
    "            input_line = input_file.readline().strip()\n",
    "            continue \n",
    "            \n",
    "\n",
    "        input_class = input_split[29]\n",
    "        if len(input_class.split(',')) > 1:\n",
    "            input_class = 'miscellaneous'  \n",
    "\n",
    "        if input_type[0:3] == 'syn':\n",
    "            if input_distance == 0:\n",
    "                if input_class == 'NM_TSS' or input_class == 'NR_TSS':                  \n",
    "                    if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                        'blank'\n",
    "                    else:\n",
    "                        ac = min(int(input_split[19].split(';')[1]),6)\n",
    "                        TSS_syn[ac-1] += 1                            \n",
    "                elif input_class == 'NM_intragenic' or input_class == 'NR_intragenic':\n",
    "                    if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                        'blank'\n",
    "                    else:\n",
    "                        ac = min(int(input_split[19].split(';')[1]),6)\n",
    "                        intra_syn[ac-1] += 1   \n",
    "\n",
    "            else: \n",
    "                if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                    'blank'\n",
    "                else:\n",
    "                    ac = min(int(input_split[19].split(';')[1]),6)\n",
    "                    noCGI_syn[ac-1] += 1     \n",
    "\n",
    "        elif input_type[0:3] == 'non':\n",
    "            if input_distance == 0:\n",
    "                if input_class == 'NM_TSS' or input_class == 'NR_TSS':                  \n",
    "                    if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                        'blank'\n",
    "                    else:\n",
    "                        ac = min(int(input_split[19].split(';')[1]),6)\n",
    "                        TSS_non[ac-1] += 1                            \n",
    "                elif input_class == 'NM_intragenic' or input_class == 'NR_intragenic':\n",
    "                    if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                        'blank'\n",
    "                    else:\n",
    "                        ac = min(int(input_split[19].split(';')[1]),6)\n",
    "                        intra_non[ac-1] += 1   \n",
    "\n",
    "            else: \n",
    "                if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                    'blank'\n",
    "                else:\n",
    "                    ac = min(int(input_split[19].split(';')[1]),6)\n",
    "                    noCGI_non[ac-1] += 1   \n",
    "\n",
    "\n",
    "        elif input_type[0:3] == 'sto':\n",
    "            if input_distance == 0:\n",
    "                if input_class == 'NM_TSS' or input_class == 'NR_TSS':                  \n",
    "                    if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                        'blank'\n",
    "                    else:\n",
    "                        ac = min(int(input_split[19].split(';')[1]),6)\n",
    "                        TSS_sto[ac-1] += 1                            \n",
    "                elif input_class == 'NM_intragenic' or input_class == 'NR_intragenic':\n",
    "                    if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                        'blank'\n",
    "                    else:\n",
    "                        ac = min(int(input_split[19].split(';')[1]),6)\n",
    "                        intra_sto[ac-1] += 1   \n",
    "\n",
    "            else: \n",
    "                if input_split[19].split(';')[0] == '.': #ExAC\n",
    "                    'blank'\n",
    "                else:\n",
    "                    ac = min(int(input_split[19].split(';')[1]),6)\n",
    "                    noCGI_sto[ac-1] += 1   \n",
    "                    \n",
    "        elif input_type[0:3] == 'unk':\n",
    "            'blank'\n",
    "        else:\n",
    "            print input_line\n",
    "            sys.exit(1)\n",
    "\n",
    "        input_line = input_file.readline().strip()\n",
    "\n",
    "output_file.write('\\t'.join(map(str,TSS_syn))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,TSS_non))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,TSS_sto))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,intra_syn))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,intra_non))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,intra_sto))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,noCGI_syn))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,noCGI_non))+'\\n')\n",
    "output_file.write('\\t'.join(map(str,noCGI_sto))+'\\n')\n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "\n",
    "#join(map(str,total_class)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "position0\n",
      "position1\n",
      "position2\n",
      "99888\n",
      "99999\n",
      "THE END\n"
     ]
    }
   ],
   "source": [
    "#Mutational signature decomposition using the 1,000 Genomes and the PCAWG projects (Af<=0.01, except Y chromosome)\n",
    "\n",
    "import sys, os, gzip\n",
    "chr_num = sys.argv[1]\n",
    "#chr_num = '1'\n",
    "output_file = file(\"Cosmic_signature_extraction_chr%s.txt\" % chr_num.replace(\"23\",\"X\"),\"w\")\n",
    "\n",
    "\n",
    "if chr_num == '23' or chr_num == 'X':\n",
    "    info_file = gzip.open(\"/home/users/jhyouk/89_backup_Workstation/jhyouk/CpG_revision/1000Genomes/ALL.chrX.phase3_shapeit2_mvncall_integrated_v1b.20130502.genotypes.vcf.gz\")\n",
    "else:\n",
    "    info_file = gzip.open(\"/home/users/jhyouk/89_backup_Workstation/jhyouk/CpG_revision/1000Genomes/ALL.chr%s.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz\" %chr_num.replace(\"23\",\"X\"))   \n",
    "\n",
    "pcawg_file = gzip.open(\"/home/users/jhyouk/89_backup_Workstation/jhyouk/CpG_revision/PCAWG-8/pcawg8.snps.indels.svs.phased.icgc.v2.vcf.gz\")\n",
    "info_line = info_file.readline().strip()\n",
    "pcawg_line = pcawg_file.readline().strip()\n",
    "while pcawg_line[0:1] == '#':\n",
    "    pcawg_line = pcawg_file.readline().strip()\n",
    "while info_line[0:1] == '#':\n",
    "    info_line = info_file.readline().strip()\n",
    "\n",
    "print 'position0'\n",
    "\n",
    "while pcawg_line:\n",
    "    p_split = pcawg_line.split('\\t')\n",
    "    if p_split[0] == chr_num.replace(\"23\",\"X\"):\n",
    "        break\n",
    "        \n",
    "    pcawg_line = pcawg_file.readline().strip()    \n",
    "\n",
    "print 'position1'\n",
    "output_file.write('#CHROM\\tPOS\\tID\\tREF\\tALT\\tAC\\tAF\\n')\n",
    "\n",
    "info_dic = {}\n",
    "m=0\n",
    "while info_line:\n",
    "    info_split = info_line.split('\\t')\n",
    "    info_chr = info_split[0]\n",
    "    info_pos = int(info_split[1])\n",
    "    info_ref = info_split[3]\n",
    "\n",
    "    info_alt = info_split[4].split(',')\n",
    "    for i in info_split[7].split(';'):\n",
    "        if i.split('=')[0] == 'AC':\n",
    "            info_ac = i.split('=')[1].split(',')      \n",
    "    \n",
    "    info_dic[info_pos] = [info_ref,info_alt,info_ac]\n",
    "    info_line = info_file.readline().strip()\n",
    "    m+=1\n",
    "    if m>100000:\n",
    "        break\n",
    "print 'position2'\n",
    "m=0\n",
    "p_dic = {}\n",
    "while pcawg_line:\n",
    "    p_split = pcawg_line.split('\\t')\n",
    "    p_chr = p_split[0]\n",
    "    if p_chr != chr_num.replace(\"23\",\"X\"):\n",
    "        break\n",
    "    p_pos = int(p_split[1])\n",
    "    p_ref = p_split[3]\n",
    "\n",
    "    p_alt = p_split[4].split(',')\n",
    "    for j in p_split[7].split(';'):\n",
    "        if j.split('=')[0] == 'AC':\n",
    "            p_ac = j.split('=')[1].split(',')     \n",
    "        \n",
    "    p_dic[p_pos] = [p_ref,p_alt,p_ac]\n",
    "    pcawg_line = pcawg_file.readline().strip() \n",
    "    m+=1\n",
    "    if m>100000:\n",
    "        break\n",
    "        \n",
    "print len(info_dic)\n",
    "print len(p_dic)\n",
    "input_min = min(min(info_dic.keys()),min(p_dic.keys()))\n",
    "input_max = max(max(info_dic.keys()),max(p_dic.keys()))\n",
    "\n",
    "ref_list = ['A','C','T','G']\n",
    "mm = 0\n",
    "for input_pos in range(input_min,input_max+1):\n",
    "    info_ref = '.'\n",
    "    info_alt = '.'\n",
    "    info_ac = 0\n",
    "    p_ref = '.'\n",
    "    p_alt = '.'\n",
    "    p_ac = 0\n",
    "    \n",
    "    if input_pos in info_dic.keys():\n",
    "        if info_dic[input_pos][0] in ref_list:\n",
    "            info_ref = info_dic[input_pos][0]\n",
    "            info_alt = info_dic[input_pos][1]\n",
    "            info_ac = info_dic[input_pos][2]\n",
    "    \n",
    "    if input_pos in p_dic.keys():\n",
    "        if p_dic[input_pos][0] in ref_list:\n",
    "            p_ref = p_dic[input_pos][0]\n",
    "            p_alt = p_dic[input_pos][1]\n",
    "            p_ac = p_dic[input_pos][2]\n",
    "    true_ref = '.'\n",
    "    if info_ac == 0 and p_ac == 0:\n",
    "        continue\n",
    "    elif info_ac > 0:\n",
    "        true_ref = info_ref\n",
    "    elif p_ac > 0:\n",
    "        true_ref = p_ref\n",
    "        \n",
    "    temp_list = [0,0,0,0]\n",
    "        \n",
    "    for ii in info_alt:\n",
    "        if ii in ref_list:\n",
    "            temp_list[ref_list.index(ii)] += int(info_ac[info_alt.index(ii)])\n",
    "    for jj in p_alt:\n",
    "        if jj in ref_list:\n",
    "            try:\n",
    "                temp_list[ref_list.index(jj)] += int(p_ac[p_alt.index(jj)])    \n",
    "            except:\n",
    "                print input_pos\n",
    "                print p_alt\n",
    "                print p_ac\n",
    "                sys.exit(1)\n",
    "    \n",
    "    for k in range(0,4):\n",
    "        if temp_list[k] > 0:\n",
    "            if float(temp_list[k])/4327.0 <=0.01:\n",
    "                output_file.write(\"%s\\t%s\\t.\\t%s\\t%s\\t%s\\t%s\\n\" % (chr_num.replace(\"23\",\"X\"),input_pos,true_ref,ref_list[k],temp_list[k],float(temp_list[k])/4327.0,))\n",
    "            mm+=1\n",
    "    if mm>1000:\n",
    "        break\n",
    "        \n",
    "output_file.close()             \n",
    "print 'THE END'        \n",
    "       "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['C>A', 'C>G', 'C>T', 'T>A', 'T>C', 'T>G']\n",
      "[['ACA', 'ACC', 'ACT', 'ACG', 'CCA', 'CCC', 'CCT', 'CCG', 'TCA', 'TCC', 'TCT', 'TCG', 'GCA', 'GCC', 'GCT', 'GCG'], ['ACA', 'ACC', 'ACT', 'ACG', 'CCA', 'CCC', 'CCT', 'CCG', 'TCA', 'TCC', 'TCT', 'TCG', 'GCA', 'GCC', 'GCT', 'GCG'], ['ACA', 'ACC', 'ACT', 'ACG', 'CCA', 'CCC', 'CCT', 'CCG', 'TCA', 'TCC', 'TCT', 'TCG', 'GCA', 'GCC', 'GCT', 'GCG'], ['ATA', 'ATC', 'ATT', 'ATG', 'CTA', 'CTC', 'CTT', 'CTG', 'TTA', 'TTC', 'TTT', 'TTG', 'GTA', 'GTC', 'GTT', 'GTG'], ['ATA', 'ATC', 'ATT', 'ATG', 'CTA', 'CTC', 'CTT', 'CTG', 'TTA', 'TTC', 'TTT', 'TTG', 'GTA', 'GTC', 'GTT', 'GTG'], ['ATA', 'ATC', 'ATT', 'ATG', 'CTA', 'CTC', 'CTT', 'CTG', 'TTA', 'TTC', 'TTT', 'TTG', 'GTA', 'GTC', 'GTT', 'GTG']]\n",
      "[[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]]\n"
     ]
    }
   ],
   "source": [
    "base3 = ['A','C','T','G']\n",
    "middle1 = ['C']\n",
    "base5 = ['A','C','T','G']\n",
    "alt1 = ['C>A','C>G','C>T']\n",
    "middle2 = ['T']\n",
    "alt2 = ['T>A','T>C','T>G']\n",
    "\n",
    "alt = ['C>A','C>G','C>T','T>A','T>C','T>G']\n",
    "context = [[],[],[],[],[],[]]\n",
    "snp_number = [[],[],[],[],[],[]]\n",
    "for l in range(0,3):\n",
    "    for i in middle1:\n",
    "        for j in base3:\n",
    "            for k in base5:\n",
    "                context[l].append(j+i+k)\n",
    "                snp_number[l].append(0)\n",
    "for l in range(3,6):\n",
    "    for i in middle2:\n",
    "        for j in base3:\n",
    "            for k in base5:\n",
    "                context[l].append(j+i+k)\n",
    "                snp_number[l].append(0)\n",
    "\n",
    "print alt\n",
    "print context\n",
    "print snp_number"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "python ../CpG/05_put5p3pbase.py final_signature_chr1.vcf 0 1 3 &> put53_1.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr2.vcf 0 1 3 &> put53_2.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr3.vcf 0 1 3 &> put53_3.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr4.vcf 0 1 3 &> put53_4.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr5.vcf 0 1 3 &> put53_5.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr6.vcf 0 1 3 &> put53_6.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr7.vcf 0 1 3 &> put53_7.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr8.vcf 0 1 3 &> put53_8.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr9.vcf 0 1 3 &> put53_9.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr10.vcf 0 1 3 &> put53_10.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr11.vcf 0 1 3 &> put53_11.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr12.vcf 0 1 3 &> put53_12.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr13.vcf 0 1 3 &> put53_13.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr14.vcf 0 1 3 &> put53_14.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr15.vcf 0 1 3 &> put53_15.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr16.vcf 0 1 3 &> put53_16.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr17.vcf 0 1 3 &> put53_17.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr18.vcf 0 1 3 &> put53_18.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr19.vcf 0 1 3 &> put53_19.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr20.vcf 0 1 3 &> put53_20.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr21.vcf 0 1 3 &> put53_21.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr22.vcf 0 1 3 &> put53_22.out\n",
      "python ../CpG/05_put5p3pbase.py final_signature_chr23.vcf 0 1 3 &> put53_23.out\n"
     ]
    }
   ],
   "source": [
    "for i in range(1,24):\n",
    "    print 'python ../CpG/05_put5p3pbase.py final_signature_chr%s.vcf 0 1 3 &> put53_%s.out' %(i,i)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pysam\n",
    "snp_type = ['C>A', 'C>G', 'C>T', 'T>A', 'T>C', 'T>G']\n",
    "context = [['ACA', 'ACC', 'ACT', 'ACG', 'CCA', 'CCC', 'CCT', 'CCG', 'TCA', 'TCC', 'TCT', 'TCG', 'GCA', 'GCC', 'GCT', 'GCG'], ['ACA', 'ACC', 'ACT', 'ACG', 'CCA', 'CCC', 'CCT', 'CCG', 'TCA', 'TCC', 'TCT', 'TCG', 'GCA', 'GCC', 'GCT', 'GCG'], ['ACA', 'ACC', 'ACT', 'ACG', 'CCA', 'CCC', 'CCT', 'CCG', 'TCA', 'TCC', 'TCT', 'TCG', 'GCA', 'GCC', 'GCT', 'GCG'], ['ATA', 'ATC', 'ATT', 'ATG', 'CTA', 'CTC', 'CTT', 'CTG', 'TTA', 'TTC', 'TTT', 'TTG', 'GTA', 'GTC', 'GTT', 'GTG'], ['ATA', 'ATC', 'ATT', 'ATG', 'CTA', 'CTC', 'CTT', 'CTG', 'TTA', 'TTC', 'TTT', 'TTG', 'GTA', 'GTC', 'GTT', 'GTG'], ['ATA', 'ATC', 'ATT', 'ATG', 'CTA', 'CTC', 'CTT', 'CTG', 'TTA', 'TTC', 'TTT', 'TTG', 'GTA', 'GTC', 'GTT', 'GTG']]\n",
    "snp_number = [[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]]\n",
    "ref_list = ['A','C','T','G']\n",
    "rev_list = ['T','G','A','C']\n",
    "\n",
    "ref_fasta = pysam.FastaFile('/home/users/jhyouk/99_reference/human/GRCh37/human_g1k_v37.fasta')\n",
    "for i in range(1,24):\n",
    "    print i\n",
    "    input_file = file(\"Cosmic_signature_extraction_chr%s.txt\",% str(i).replace(\"23\",\"X\"))\n",
    "    input_line = input_file.readline().strip()\n",
    "    input_line = input_file.readline().strip()\n",
    "    \n",
    "    while input_line:\n",
    "        input_split = input_line.split('\\t')\n",
    "        input_chr = input_split[0]\n",
    "        input_pos = int(input_split[1])\n",
    "        input_ref = input_split[3]\n",
    "        input_alt = input_split[4]\n",
    "        \n",
    "        temp_base5 = ref_fasta.fetch(input_chr,input_pos-2,input_pos-1)\n",
    "        temp_base3 = ref_fasta.fetch(input_chr,input_pos,input_pos+1)\n",
    "        \n",
    "        if input_ref == 'C' or input_ref == 'T':\n",
    "            base5 = temp_base5\n",
    "            base0 = input_ref\n",
    "            base0_alt = input_alt\n",
    "            base3 = temp_base3\n",
    "        else:\n",
    "            base5 = rev_list[ref_list.index(temp_base3)]\n",
    "            base0 = rev_list[ref_list.index(input_ref)]\n",
    "            base0_alt = rev_list[ref_list.index(input_alt)]\n",
    "            base3 = rev_list[ref_list.index(temp_base5)]\n",
    "        \n",
    "        input_snp_type = base0 + '>' + base0_alt\n",
    "        input_index = snp_type.index(input_snp_type)\n",
    "        input_context = base5 + base0 + base3\n",
    "        context_index = context[input_index].index(input_context)\n",
    "        snp_number[input_index][context_index] += 1\n",
    "    \n",
    "        input_line = input_file.readline().strip()\n",
    "    input_file.close()\n",
    "    \n",
    "output_file = file(\"Cosmic_signature_96contexts.txt\",\"w\")\n",
    "for i in snp_type:\n",
    "    for j in range(0,16):\n",
    "        output_file.write(\"%s\\t%s\\t%s\\n\" %(i,context[snp_type.index(i)][j],snp_number[snp_type.index(i)][j]))\n",
    "        \n",
    "output_file.close()\n",
    "print 'THE END'\n",
    "    "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
